examples/shader/post_processing.rs

```diff
@@ -1,188 +1,392 @@
-//! A custom post processing effect, using two cameras, with one reusing the render texture of the first one.
-//! Here a chromatic aberration is applied to a 3d scene containing a rotating cube.
-//! This example is useful to implement your own post-processing effect such as
-//! edge detection, blur, pixelization, vignette... and countless others.
+//! This example shows how to create a custom render pass that runs after the main pass
+//! and reads the texture generated by the main pass.
+//!
+//! The example shader is a very simple implementation of chromatic aberration.
+//!
+//! This is a fairly low level example and assumes some familiarity with rendering concepts and wgpu.
 
 use bevy::{
-    core_pipeline::clear_color::ClearColorConfig,
+    core_pipeline::{
+        clear_color::ClearColorConfig, core_3d,
+        fullscreen_vertex_shader::fullscreen_shader_vertex_state,
+    },
     prelude::*,
-    reflect::TypeUuid,
     render::{
-        camera::RenderTarget,
+        extract_component::{
+            ComponentUniforms, ExtractComponent, ExtractComponentPlugin, UniformComponentPlugin,
+        },
+        render_graph::{Node, NodeRunError, RenderGraphApp, RenderGraphContext},
         render_resource::{
-            AsBindGroup, Extent3d, ShaderRef, TextureDescriptor, TextureDimension, TextureFormat,
-            TextureUsages,
+            BindGroupDescriptor, BindGroupEntry, BindGroupLayout, BindGroupLayoutDescriptor,
+            BindGroupLayoutEntry, BindingResource, BindingType, CachedRenderPipelineId,
+            ColorTargetState, ColorWrites, FragmentState, MultisampleState, Operations,
+            PipelineCache, PrimitiveState, RenderPassColorAttachment, RenderPassDescriptor,
+            RenderPipelineDescriptor, Sampler, SamplerBindingType, SamplerDescriptor, ShaderStages,
+            ShaderType, TextureFormat, TextureSampleType, TextureViewDimension,
         },
+        renderer::{RenderContext, RenderDevice},
         texture::BevyDefault,
-        view::RenderLayers,
+        view::{ExtractedView, ViewTarget},
+        RenderApp,
     },
-    sprite::{Material2d, Material2dPlugin, MaterialMesh2dBundle},
 };
 
 fn main() {
     App::new()
-        .add_plugins(DefaultPlugins)
-        .add_plugin(Material2dPlugin::<PostProcessingMaterial>::default())
+        .add_plugins(DefaultPlugins.set(AssetPlugin {
+            // Hot reloading the shader works correctly
+            watch_for_changes: true,
+            ..default()
+        }))
+        .add_plugin(PostProcessPlugin)
         .add_systems(Startup, setup)
-        .add_systems(Update, main_camera_cube_rotator_system)
+        .add_systems(Update, (rotate, update_settings))
         .run();
 }
 
-/// Marks the first camera cube (rendered to a texture.)
-#[derive(Component)]
-struct MainCube;
+/// It is generally encouraged to set up post processing effects as a plugin
+struct PostProcessPlugin;
```

Comment:
```suggestion
struct PostProcessPlugin;

```

Reply:
I'm not sure I understand what the suggestion does. It just looks like nothing in the preview.

Reply:
Oh, it's adding a newline